x-pack/plugins/security/server/plugin.ts

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { first, map } from 'rxjs/operators';
+import {
+  ClusterClient,
+  CoreSetup,
+  KibanaRequest,
+  Logger,
+  PluginInitializerContext,
+  RecursiveReadonly,
+  deepFreeze,
+} from '../../../../src/core/server';
+import { XPackInfo } from '../../../legacy/plugins/xpack_main/server/lib/xpack_info';
+import { AuthenticatedUser } from '../common/model';
+import { Authenticator, setupAuthentication } from './authentication';
+import { createConfig$ } from './config';
+
+/**
+ * Describes a set of APIs that is available in the legacy platform only and required by this plugin
+ * to function properly.
+ */
+export interface LegacyAPI {
```

Comment:
__note:__ that's the only thing we drag from the LP. I tried to narrow down required "features" as much as possible so that we see what is actually used from X-Pack and others.